engines/lab/engine.cpp

```diff
@@ -0,0 +1,1190 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+/*
+ * This code is based on Labyrinth of Time code with assistance of
+ *
+ * Copyright (c) 1993 Terra Nova Development
+ * Copyright (c) 2004 The Wyrmkeep Entertainment Co.
+ *
+ */
+
+#include "lab/lab.h"
+
+#include "lab/anim.h"
+#include "lab/dispman.h"
+#include "lab/eventman.h"
+#include "lab/image.h"
+#include "lab/intro.h"
+#include "lab/labsets.h"
+#include "lab/music.h"
+#include "lab/processroom.h"
+#include "lab/resource.h"
+#include "lab/tilepuzzle.h"
+#include "lab/utils.h"
+
+namespace Lab {
+
+// LAB: Labyrinth specific code for the special puzzles
+#define SPECIALLOCK         100
+#define SPECIALBRICK        101
+#define SPECIALBRICKNOMOUSE 102
+
+enum Items {
+	kItemHelmet = 1,
+	kItemBelt = 3,
+	kItemPithHelmet = 7,
+	kItemJournal = 9,
+	kItemNotes = 12,
+	kItemWestPaper = 18,
+	kItemWhiskey = 25,
+	kItemLamp = 27,
+	kItemMap = 28,
+	kItemQuarter = 30
+};
+
+#define kCondLampOn         151
+#define kCondBeltGlowing     70
+#define kCondUsedHelmet     184
+
+enum Monitors {
+	kMonitorMuseum = 71,
+	kMonitorGramophone = 72,
+	kMonitorUnicycle = 73,
+	kMonitorStatue = 74,
+	kMonitorTalisman = 75,
+	kMonitorLute = 76,
+	kMonitorClock = 77,
+	kMonitorWindow = 78,
+	//kMonitorBelt = 79,
+	kMonitorLibrary = 80,
+	kMonitorTerminal = 81
+	//kMonitorLevers = 82
+};
+
+enum MainButtons {
+	kButtonPickup,
+	kButtonUse,
+	kButtonOpen,
+	kButtonClose,
+	kButtonLook,
+	kButtonInventory,
+	kButtonLeft,
+	kButtonForward,
+	kButtonRight,
+	kButtonMap
+};
+
+enum AltButtons {
+	kButtonMainDisplay,
+	kButtonSaveLoad,
+	kButtonUseItem,
+	kButtonLookAtItem,
+	kButtonPrevItem,
+	kButtonNextItem,
+	kButtonBreadCrumbs,
+	kButtonFollowCrumbs
+};
+
+static char initcolors[] = { '\x00', '\x00', '\x00', '\x30',
```

Comment:
Should be "initColors", I think. One might also consider make the base type `byte` and to use 0x## to make it more clear that this is no character data.